lighthouse/routes/common/plates.py

```diff
@@ -24,6 +34,55 @@
 logger = logging.getLogger(__name__)
 
 
+def get_control_locations() -> FlaskResponse:
+    """
+    Get the POST body and validate the presence of fields
+    Then call Sequencescape, to get sample information for the plate
+    Get the control locations for the plate, from sample infomation
+    Return response to Beckman robot
+
+    Returns:
+        FlaskResponse: json body containing the plate barcode, and +ve/-ve control locations
+    """
+
+    logger.info("Attemping to fetching control locations for barcode")
+
+    # Validate presence of user, robot, barcode
+    barcode = None
+    user = None
+    robot = None
+
+    if (request_json := request.get_json()) is not None:
+        barcode = request_json.get(ARG_BARCODE)
+        user = request_json.get(ARG_USER)
+        robot = request_json.get(ARG_ROBOT_SERIAL)
+
+    # Return 400 if POST body fails validation
+    if barcode is None or user is None or robot is None:
+        return bad_request(f"POST request needs '{ARG_BARCODE}', '{ARG_USER}' and '{ARG_ROBOT_SERIAL}' in body")
+
+    # Send GET to Sequencescape, to return all samples for the barcode
+    # TODO (DPL-572): This could be moved to another file, possibly an Sequencescape API wrapper file?
+    try:
+        ss_response = get_from_ss_plates_samples_info(barcode)
+    except Exception as exc:
+        return internal_server_error(str(exc))
+
+    # TODO (DPL-572): Maybe don't need to pass in barcode? Barcode is needed for error messages
+    # but maybe that could be handled elsewhere
+    # TODO (DPL-572): Possibly change method name
```

Comment:
There are a few more TODOs here and another below.  These should either reference a new ticket where they'll be addressed or should be removed if they've already been addressed.  The idea is that we can see more detail about the TODO in the ticket and can schedule in doing it.  TODOs in code are a form of litter that stays for years and years.

Reply:
Checked and removed all DPL-572 TODO comments